package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -0,0 +1,943 @@
+                            #             dist_shift,
+                            #             grid_shift[XX], grid_shift[YY], grid_shift[ZZ],
+                            #             np.sqrt(np.sum(grid_shift**2))
+                            #         ))
+
+
+                            for j in range(self.grid.nbeads[cellindex_adjacent]):
+                                bid = self.grid.beadids[cellindex_adjacent * self.grid.nbeads_per_cell + j]
+
+                                if checked[bid] != 0:
+                                    continue
+
+                                d2 = self.box.fast_distance2(&self.coords_bbox[current_beadid, XX], &self.coords_bbox[bid, XX])
+
+                                # if self.debug:
+                                #     self.grid.cellid2cellxyz(cellindex, debug_cellxyz)
+                                #     with gil:
+                                #         print(
+                                #             "Beads #{} (cell[{},{},{}]-coords[{:.3f},{:.3f},{:.3f}]) and #{} (cell[{},{},{}]-coords[{:.3f},{:.3f},{:.3f}]) are tested (d2={:.3f})".format(
+                                #                 current_beadid,
+                                #                 cellxyz[XX], cellxyz[YY], cellxyz[ZZ],
+                                #                 self.coords_bbox[current_beadid, XX],
+                                #                 self.coords_bbox[current_beadid, YY],
+                                #                 self.coords_bbox[current_beadid, ZZ],
+                                #                 bid,
+                                #                 debug_cellxyz[XX], debug_cellxyz[YY], debug_cellxyz[ZZ],
+                                #                 self.coords_bbox[bid, XX], self.coords_bbox[bid, YY],
+                                #                 self.coords_bbox[bid, ZZ],
+                                #                 d2))
+
+                                if d2 < cutoff2:
+
+                                    if d2 < EPSILON:
+                                        continue
+
+                                    # if self.debug and debug:
+                                    #     self.grid.cellid2cellxyz(cellindex, debug_cellxyz)
+                                    #     with gil:
+                                    #         self.box.fast_pbc_dx(&self.coords[current_beadid, XX], &self.coords[bid, XX], dx)
+                                    #         dx_py = np.array([dx[XX], dx[YY], dx[ZZ]])
+                                    #         print("FastNS:       \_ Neighbor found:  bead#{} (cell[{},{},{}]) -> dx={} -> d={:.3f}".format(bid, debug_cellxyz[XX], debug_cellxyz[YY], debug_cellxyz[ZZ], dx_py, np.sqrt(d2)))
+                                    if results.add_neighbors(current_beadid, bid, d2) == 0:
+                                        memory_error = True

```

Comment:
is this really a memory error?

If we only change this here do we need all the checks?

Reply:
Yes, it is a memory error: As raised exceptions are ignored in nogil function, the underlying PyMem_Realloc failure is passed as a return code of zero